router/worker.go

```diff
@@ -979,6 +985,15 @@ func (w *worker) postStatusOnResponseQ(respStatusCode int, payload json.RawMessa
 		}
 	}
 
+	inputPayload := payload
+	switch errorAt {
+	case routerutils.ERROR_AT_TF:
+		inputPayload = destinationJobMetadata.JobT.EventPayload
+		status.ErrorResponse = misc.UpdateJSONWithNewKeyVal(status.ErrorResponse, "failureStage", "RudderStack Transformation Error")
+	default: // includes ERROR_AT_DEL, ERROR_AT_CUST
+		status.ErrorResponse = misc.UpdateJSONWithNewKeyVal(status.ErrorResponse, "failureStage", "Destination Error")
+	}
+
 	status.ErrorResponse = routerutils.EnhanceJSON(status.ErrorResponse, "firstAttemptedAt", firstAttemptedAtTime.Format(misc.RFC3339Milli))
 	status.ErrorResponse = routerutils.EnhanceJSON(status.ErrorResponse, "content-type", respContentType)
```

Comment:
Would it make sense to always include `errorAt` as part ot the `ErrorResponse` to aid with debugging?

Reply:
In the other cases, the job states and `reason` inform us about why that job failed.

Reply:
I mean to say all cases are already pretty well covered to inform an observer as to why jobs failed at a particular stage.